apigee_edge.routing.yml

```diff
@@ -5,7 +5,6 @@ apigee_edge.error_page:
     _title_callback: '\Drupal\apigee_edge\Controller\ErrorPageController::getPageTitle'
   requirements:
     _permission: 'access content'
-    _user_is_logged_in: 'TRUE'
```

Comment:
We initially added this to ensure that API communication error page is only visible to logged in user, it is not going to be indexed by search engines, etc. However, it could happen indeed that some content from Apigee Edge gets displayed to anonymous users as well and in case of a communication issue they should have access to this error page.

Question: can we ensure somehow that this page is not indexed by robots? Or is it already working like this?

Reply:
If this page needs a logged in user, then the subrequest will fail in the exception handler.

This page won't be linked directly, so I don't think that it will get indexed ever... but I'll take a look if the default Drupal error pages do something with search engine indexing.

Reply:
AFAIK if the page is not linked and/or if the path is not in sitemap.xml then it cannot be indexed. I don't know what happens if an exception is thrown when a crawler fetches one of the affected pages (if there is any for anonymous users). In my opinion, a comment is enough here. In this special case if the page is indexed then the maintainers/developers should add the path to the site's robots.txt.

Reply:
We just need an event subscriber that adds the NOINDEX header to the api-communication-error page:
https://cgit.drupalcode.org/http_response_headers/tree/src/EventSubscriber/AddHTTPHeaders.php?id=de12feb#n71

Reply:
Default Drupal error pages are not mentioned in the robots.txt or set special headers, since they never linked directly.